### ISSUE TITLE
Issue converting MHGU (switch) -> MHXX (3DS)

### ISSUE THREAD

**Issue by Ninjaskill** [Dawnshifter/MHXXSwitchSaveEditor#8]
Hi,

After using this app I ran into the following issue.
3DS MHXX -> MHGU Switch works perfect. I loaded the file in-game was able to save without any issues only crashes when you open DLC events.

MHGU Switch -> 3DS
This is where the problem begins. First off the file size is rather strange it stays the same file size and the switch save file. I think this might be part of the problem. Furthermore once the Switch save has been converted to 3DS it's no longer openable or reconizeable by the save editor. Citra is able to load the file but not able to save but for some reason it saves fine on a real 3ds though.

**Comment by ProudCatz** [comment]
The save files of MHXX and MHGU may be different,two games cannot be connected,so the data might be different,Unable to convert

**Comment by Ninjaskill** [comment]
> The save files of MHXX and MHGU may be different,two games cannot be connected,so the data might be different,Unable to convert

No, Switch -> 3DS conversion definitely works.  After some trial and error I have also figured out the problem. Apparently, the switch file has extra reserved data at the end of the file and when you convert to 3DS with the save editor it doesn't remove the excessive bit which makes the app and Citra think the file invalid.

The fix to the problem I have found and confirmed working is by opening the file in a hex editor and remove the excessive lines. After the last hex id you need to have 32 lines and remove any excessive lines. See pic for reference

So, the problem is with the file size. 3DS system file size is 4.616kb while the switch is 5.039kb when converting to 3DS the filesize stays 5.039kb which is causing the save editor not to be able to read the file anymore when you're trying to open it. Also causes Citra having saving issues thinking the file is corrupted when trying to save.

<img width="235" alt="Knipsel" src="https://user-images.githubusercontent.com/10984882/75338846-b2b51180-588f-11ea-8f32-bedd89afba87.PNG">

**Comment by blagapar** [comment]
Alright, so I tried your thing with a lot of save files, and it worked with many of them, but not with all. But I still managed to FIX it, here's how. 
As you said, 3ds needs a 4.616kb save file, so : 
- I created the "switch to 3ds" converted save file using mhgu save editor
- I opened the file in HxD
- Then you need to go in "view > offset base > choose Decimal". Also keep the "bytes per row" at 16
- Now, under "Offset (d)" you have the find the value "04726144" and remove all lines under it, even if they have codes. Also make sure to delete the 4 last HEX values on the line "04726144"

And that's it.

**Comment by jackology** [comment]
i also face this problem and i delete all after the line "04726144" 
but still can't open the file converted to 3ds save by the mhgu editer or mhxx editer,any things which can i do to fix this problem?

here is the erro photo
i think it maybe due to some item was not exist in 3ds version?
![image](https://user-images.githubusercontent.com/52628716/107889882-72431d80-6f50-11eb-93f8-18e5380a8a96.jpg)

**Comment by jackology** [comment]
> i also face this problem and i delete all after the line "04726144"
> but still can't open the file converted to 3ds save by the mhgu editer or mhxx editer,any things which can i do to fix this problem?
> 
> here is the erro photo
> i think it maybe due to some item was not exist in 3ds version?
> ![image](https://user-images.githubusercontent.com/52628716/107889882-72431d80-6f50-11eb-93f8-18e5380a8a96.jpg)

finally i cut the file size with exactly with 4726152, and it can be read by mhxx editer but when i convert it to switch version it still convert to mhgu save verion, which can't convert it to mhxx version(solved)
(thanks i solve the problem finally , the file show with mhgu but which can be read by mhxx. so i transfer successfully.)
8

**Comment by 007FFF** [comment]
> Alright, so I tried your thing with a lot of save files, and it worked with many of them, but not with all. But I still managed to FIX it, here's how. As you said, 3ds needs a 4.616kb save file, so :
> 
> * I created the "switch to 3ds" converted save file using mhgu save editor
> * I opened the file in HxD
> * Then you need to go in "view > offset base > choose Decimal". Also keep the "bytes per row" at 16
> * Now, under "Offset (d)" you have the find the value "04726144" and remove all lines under it, even if they have codes. Also make sure to delete the 4 last HEX values on the line "04726144"
> 
> And that's it.

Hi, do you know how to do this when converting MHXX Switch to 3DS?

**Comment by 007FFF** [comment]
Figured it out. To convert XX Switch to 3DS, you gotta convert to 3DS using the save editor, put the file into HxD, go into decimal view, scroll all the way down, and add enough 00's until you're missing 4 HEX values on line 04726144. I don't remember the exact amount, but just count how many you're missing, and it'll work.

**Comment by 007FFF** [comment]
Also, when converting from 3DS -> Switch, is it normal for the 'Loading..' prompt to be stuck on "Loading your save file.. Please wait.."?

**Comment by Klaus2442** [comment]
Hey, I'm having problems converting MHXX (3ds) to MHGU (Switch), could someone tell me how to do it? I'm kinda lost

**Comment by JoJoelix** [comment]
> Alright, so I tried your thing with a lot of save files, and it worked with many of them, but not with all. But I still managed to FIX it, here's how. As you said, 3ds needs a 4.616kb save file, so :
> 
> * I created the "switch to 3ds" converted save file using mhgu save editor
> * I opened the file in HxD
> * Then you need to go in "view > offset base > choose Decimal". Also keep the "bytes per row" at 16
> * Now, under "Offset (d)" you have the find the value "04726144" and remove all lines under it, even if they have codes. Also make sure to delete the 4 last HEX values on the line "04726144"
> 
> And that's it.

son of a bitch, i did it, it worked! thank you!
